### DISCUSSION TITLE
How to setup docker for QNAP Container

### DISCUSSION THREAD

**Discussion by jucytec** [mrlt8/docker-wyze-bridge#80] [Q&A]
Very Noob Question.

qnap NAS has a container station that can run docker.  I saw that this was posted on the docker store... I installed it but I have NO idea how to edit the config file with the password and username.

I've tried sending docker run -p 1935:1935 -p 8554:8554 -p 8888:8888 -e WYZE_EMAIL='email@email.com' -e WYZE_PASSWORD='pass'  mrlt8/wyze-bridge which sends me a authentication text so it feels like its working... BUT  on that same ssh session if i echo the sms auth code nothing happens.

docker exec -it wyze-bridge sh -c 'echo "SMSauthcode" > /tokens/mfa_token'

**Comment by mrlt8** [comment]
The container is meant to be run as a service, so you'll need to add the code using another ssh session.

**Reply by jucytec** [reply]
Ah, ok thank you,!  i see that it assigns a random container name so i was able to execute a new command and it is now working i think...  what viewer do i use on Windows or Android to view the RTSP feed?

**Reply by mrlt8** [reply]
Not sure what player to use on windows (VLC maybe?), but you can test the the HLS stream directly in a browser with `http://server-ip:8888/camera-nickname`

**Reply by jucytec** [reply]
Yes, that worked, I can stream using http, but when try to add the RTSP feed to the qnap server or vlc, i just can an error.

i was hoping to use the RTSP feed to record to my NAS but so far no luck.  I was using rtsp://localip:8554/cam1 

I'm not sure if i did anything wrong.  Thanks for helping out.

**Reply by mrlt8** [reply]
Strange.. maybe something else is already using that port?
Might be able to change the port with `-e RTSP_RTSPADDRESS=:8555`

**Reply by sjtsnix** [reply]
> The container is meant to be run as a service, so you'll need to add the code using another ssh session.

Hello,
New to Docker Containers on my QNAP NAS.  

I've established an SSH session to my QNAP NAS

I then changed to the container using the device ID of the container: 
**docker exec -it d63d17ac30b7 /bin/bash**

I then copy /pasted the following using my email address and wyze password at the SSH command line: 

**docker run \
  -e WYZE_EMAIL=you@email.com \
  -e WYZE_PASSWORD=yourpassw0rd \
  -p 8888:8888 mrlt8/wyze-bridge:latest**
 
There was no feedback saying 'successful' or not. (new to this, so don't know what I should see occur here)

Should it be typed (copy/pasted) exactly as you list here? 

at the SSH command line window, this continues to scroll in that window: 

**2022/07/09 20:35:14 [WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2022/07/09 20:35:14 [WyzeBridge] 'NoneType' object has no attribute 'encode'**

Thank you for any assistance you can provide me as I attempt to get this to work.

SJ

**Reply by mrlt8** [reply]
@sjtsnix are you trying to enter a two-factor authentication code? Otherwise, you shouldn't need to issue any commands inside the container once it's up.

Also, if you're using the docker run command, be sure to have the entire command on a single line _or_ include a `\` before the line break.
